# Selecting samples for Synthetic data generation

## Importing libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
import os
import sys
from pathlib import Path

path = Path(os.path.dirname(os.getcwd()))
path = str(path)
print(path)
sys.path.insert(1, path)

/Users/saideepbunny/Projects/Application_Ranking_System


## Reading the data

In [3]:
df = pd.read_csv(f"{path}/data/kaggle_linkedin_jd_skills_dataset/merged_selected_jd.csv")
df

job_link  \
0       https://www.linkedin.com/jobs/view/independent...   
1       https://www.linkedin.com/jobs/view/chef-de-cui...   
2       https://www.linkedin.com/jobs/view/cheese-spec...   
3       https://www.linkedin.com/jobs/view/retail-dist...   
4       https://www.linkedin.com/jobs/view/travel-rn-l...   
...                                                   ...   
125375  https://uk.linkedin.com/jobs/view/production-p...   
125376  https://www.linkedin.com/jobs/view/store-manag...   
125377  https://www.linkedin.com/jobs/view/sales-repre...   
125378  https://www.linkedin.com/jobs/view/ft-center-s...   
125379  https://www.linkedin.com/jobs/view/associate-a...   

                  last_processed_time got_summary got_ner is_being_worked  \
0       2024-01-21 07:40:00.308133+00           t       t               f   
1       2024-01-21 07:12:35.170823+00           t       t               f   
2       2024-01-21 06:01:14.692423+00           t       t               f   
3       2024-01-21 07:12:35.189652+00           t       t               f   
4       2024-01-21 07:12:37.537623+00           t       t               f   
...                               ...         ...     ...             ...   
125375  2024-01-20 04:01:08.913697+00           t       t               f   
125376  2024-01-20 05:12:47.255315+00           t       t               f   
125377  2024-01-20 13:46:06.129745+00           t       t               f   
125378  2024-01-20 13:46:10.289403+00           t       t               f   
125379   2024-01-20 15:20:54.32033+00           t       t               f   

                                 job_title                          company  \
0            Independent Real Estate Agent       Howard Hanna | Rand Realty   
1                          Chef de Cuisine               Goodwin Recruiting   
2                        Cheese Specialist                          Safeway   
3       RETAIL DISTRICT MANAGER UNASSIGNED                   Dollar General   
4               Travel RN - Long Term Care                CoreMedical Group   
...                                    ...                              ...   
125375                  Production Planner  Eat Real + PROPER | WARP Snacks   
125376           Store Manager in Training              O'Reilly Auto Parts   
125377                Sales Representative  American Fidelity Sales Careers   
125378         FT Center Store Manager (H)                        Food Lion   
125379           Associate Account Manager                          Lockton   

                             job_location  first_seen    search_city  \
0                    Englewood Cliffs, NJ  2024-01-16      Pinehurst   
1                          Brookfield, WI  2024-01-14      Milwaukee   
2                          Mount Airy, MD  2024-01-14      Frederick   
3                             Newport, NH  2024-01-14  New Hampshire   
4                          Barnstable, MA  2024-01-14     Barnstable   
...                                   ...         ...            ...   
125375  Nuneaton, England, United Kingdom  2024-01-14       Hastings   
125376                        Concord, NH  2024-01-14        Laconia   
125377                        Altoona, PA  2024-01-14   Pennsylvania   
125378                       Powhatan, VA  2024-01-14       Abingdon   
125379                    Minneapolis, MN  2024-01-15    Lake Forest   

        search_country             search_position   job_level job_type  \
0        United States           Real-Estate Clerk  Mid senior   Onsite   
1        United States                   Sous Chef  Mid senior   Onsite   
2        United States  Special Effects Specialist  Mid senior   Onsite   
3        United States         Pressure Supervisor  Mid senior   Onsite   
4        United States                  Nurse Head  Mid senior   Onsite   
...                ...                         ...         ...      ...   
125375  United Kingdom                   Scheduler  

In [4]:
df.isna().sum()

job_link                 0
last_processed_time      0
got_summary              0
got_ner                  0
is_being_worked          0
job_title                0
company                  3
job_location             2
first_seen               0
search_city              0
search_country           0
search_position          0
job_level                0
job_type                 0
job_skills             216
job_summary              0
id                       0
summary_len              0
dtype: int64

In [5]:
#drop those records with null values in job_skills column
df = df.dropna(subset=['job_skills']).copy()

## Examining job summaries

### All job summaries

In [6]:
print(df.loc[0]['job_summary'])

Who We Are
Rand Realty is a family-owned brokerage born in 1984 when our founder, Marsha opened her new real estate office in a small town in Hudson Valley. Her vision was to build a brokerage strongly devoted to its agents and deeply committed to the community combining expertise and eye for detail with heart, soul and kindness. More than 35 years later, we've grown to more than 1,000 premier local agents with offices throughout Northern New Jersey, Westchester, Connecticut, The Bronx, Rockland, Dutchess, Putnam and Orange counties. Despite our growth or really, by virtue of it and the great people who have joined and represent our brand, the characteristics that defined us then, continue to dictate and delineate who we are.
Today, Howard Hanna | Rand Realty is the preeminent real estate brokerage in New York, New Jersey and Connecticut, widely recognized as an industry leader for its agent training, career planning, technological innovation and market expertise. Build your career wit

In [7]:
print(df.loc[1]['job_summary'])

To Apply for this Job Click Here
Midwest Hospitality Group seeks a Chef de Cuisine to lead its Culinary Team. The ideal candidate will possess experience as a Chef de Cuisine and/or Executive Sous Chef in Independent Restaurants or Hotel Food & Beverage operations. Are you looking for a new challenge with a stable Company whose Culture is focused on providing the Guest with the highest quality Food and Service and offers growth opportunities? If so, please submit your resume today!
Benefits
Salary - $80,000 to $90,000
Health insurance
Life insurance
Free Parking
Shift Meals
PTO immediately
Company Discounts
Responsibilities
Ensure proper team coverage to execute a high level of production while keeping payroll costs in line
Oversee ordering, receiving, storage (including temperature-setting) usage, and rotation of food
Works with Maintenance in the repair of equipment and preventative maintenance programs
Assists with menu development and engineering
Leads culinary staff in the proper 

We can see from the above sample that few of the job summaries specifically have "Responsibilities" in the text. Since it is not possible to evaluate all the ~125k records to see if the job summary includes Responsibilities, we can simply filter the records based on the existence of the string "Responsibilities".

### Records containing "Responsibilities" in job summary

In [8]:
resp_df = df[df['job_summary'].str.contains("Responsibilities")]
resp_df

job_link  \
1       https://www.linkedin.com/jobs/view/chef-de-cui...   
7       https://www.linkedin.com/jobs/view/rad-tech-at...   
8       https://www.linkedin.com/jobs/view/retail-stoc...   
9       https://www.linkedin.com/jobs/view/grooming-sa...   
10      https://ca.linkedin.com/jobs/view/senior-produ...   
...                                                   ...   
125374  https://www.linkedin.com/jobs/view/quality-aud...   
125375  https://uk.linkedin.com/jobs/view/production-p...   
125377  https://www.linkedin.com/jobs/view/sales-repre...   
125378  https://www.linkedin.com/jobs/view/ft-center-s...   
125379  https://www.linkedin.com/jobs/view/associate-a...   

                  last_processed_time got_summary got_ner is_being_worked  \
1       2024-01-21 07:12:35.170823+00           t       t               f   
7       2024-01-19 14:39:33.898387+00           t       t               f   
8       2024-01-21 07:43:49.202444+00           t       t               f   
9       2024-01-21 07:43:50.084041+00           t       t               f   
10      2024-01-21 07:43:52.439593+00           t       t               f   
...                               ...         ...     ...             ...   
125374  2024-01-20 04:01:06.622679+00           t       t               f   
125375  2024-01-20 04:01:08.913697+00           t       t               f   
125377  2024-01-20 13:46:06.129745+00           t       t               f   
125378  2024-01-20 13:46:10.289403+00           t       t               f   
125379   2024-01-20 15:20:54.32033+00           t       t               f   

                          job_title                           company  \
1                   Chef de Cuisine                Goodwin Recruiting   
7                          Rad Tech  Virginia Mason Franciscan Health   
8         Retail Stocking Team Lead           Burlington Stores, Inc.   
9            Grooming Salon Manager                             Petco   
10           Senior Product Manager                       Robert Half   
...                             ...                               ...   
125374              Quality Auditor                           ANGARAI   
125375           Production Planner   Eat Real + PROPER | WARP Snacks   
125377         Sales Representative   American Fidelity Sales Careers   
125378  FT Center Store Manager (H)                         Food Lion   
125379    Associate Account Manager                           Lockton   

                             job_location  first_seen   search_city  \
1                          Brookfield, WI  2024-01-14     Milwaukee   
7                             Seattle, WA  2024-01-13        Summit   
8                            Chandler, AZ  2024-01-16      Chandler   
9                            Sarasota, FL  2024-01-16      Sarasota   
10               Calgary, Alberta, Canada  2024-01-14       Calgary   
...                                   ...         ...           ...   
125374                     Washington, DC  2024-01-14    Washington   
125375  Nuneaton, England, United Kingdom  2024-01-14      Hastings   
125377                        Altoona, PA  2024-01-14  Pennsylvania   
125378                       Powhatan, VA  2024-01-14      Abingdon   
125379                    Minneapolis, MN  2024-01-15   Lake Forest   

        search_country             search_position   job_level job_type  \
1        United States                   Sous Chef  Mid senior   Onsite   
7        United States                 Radiologist  Mid senior   Onsite   
8        United States     Pattern-Shop Supervisor  Mid senior   Onsite   
9        United States   Quality Assurance Monitor  Mid senior   Onsite   
10              Canada                        Tier  Mid senior   Onsite   
...                ...                         ...         ...      ...   
125374   United States          Standards Engineer  Mid senior   Onsite   
125375  United Kingdom                   Scheduler  Mid senio

In [9]:
for i in range(10):
    print(f"Record {i+1}:")
    print(resp_df.iloc[i]['job_summary'])
    print("\n================================================================================\n")

Record 1:
To Apply for this Job Click Here
Midwest Hospitality Group seeks a Chef de Cuisine to lead its Culinary Team. The ideal candidate will possess experience as a Chef de Cuisine and/or Executive Sous Chef in Independent Restaurants or Hotel Food & Beverage operations. Are you looking for a new challenge with a stable Company whose Culture is focused on providing the Guest with the highest quality Food and Service and offers growth opportunities? If so, please submit your resume today!
Benefits
Salary - $80,000 to $90,000
Health insurance
Life insurance
Free Parking
Shift Meals
PTO immediately
Company Discounts
Responsibilities
Ensure proper team coverage to execute a high level of production while keeping payroll costs in line
Oversee ordering, receiving, storage (including temperature-setting) usage, and rotation of food
Works with Maintenance in the repair of equipment and preventative maintenance programs
Assists with menu development and engineering
Leads culinary staff in t

## Statistics for summary length and skill count

In [10]:
#Splitting job skills by comma to get skill count
resp_df.loc[:,'skill_count'] = resp_df['job_skills'].apply(lambda x: len(x.split(","))).copy()


/var/folders/t5/tflj5hjs197_vv37fw2sdhdr0000gn/T/ipykernel_11981/3804758628.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  resp_df.loc[:,'skill_count'] = resp_df['job_skills'].apply(lambda x: len(x.split(","))).copy()


In [11]:
#getting max summary_len for each job title using groupby
resp_df[['job_title', 'summary_len', 'skill_count']].groupby(['job_title']).aggregate({"summary_len":"max", "skill_count":"max"})

summary_len  skill_count
job_title                                                                   
2024 - Culinary Professionals - Assistant Kitch...         5863           69
2024 - Culinary Professionals - Dining Hall Ser...         5593           69
AE - Sales Leader (Full-Time) - US                         4997           46
ARNP                                                       6175           56
ASSISTANT MANAGER                                          5375           99
...                                                         ...          ...
Workshop Controller                                        4703           32
X-Ray Technologist                                         6213           41
Xfinity Retail Sales Consultant                            4972           32
Yard Associate (SkillBridge)                               2292           48
Yoga Instructor                                            3782           30

[1168 rows x 2 columns]

In [12]:
print("Number of unique job titles:", resp_df['job_title'].nunique())

Number of unique job titles: 1168


The plan is to obtain 1500 records for generating synthetic resumes. Therefore, it would be better to choose the job_summary that has the most information about the job. So, for each job title, I will select the record with max. summary_len. That makes 1168 records. For the rest of the records, I will select some famous job titles like Data Engineer, Data Scientist, etc., that have the highest number of skill_count

## Final Sampling

### Sample 1 Job per job title

In [13]:
#Getting job that has maximum length job_summary for each job title
temp = resp_df[resp_df['skill_count'] >=30].copy()
resp_df_max = temp.loc[temp.groupby('job_title')['summary_len'].idxmax()].reset_index(drop=True).copy()
resp_df_max = resp_df_max.drop_duplicates(keep='first', subset=['job_summary'])
resp_df_max

job_link  \
0     https://www.linkedin.com/jobs/view/2024-culina...   
1     https://www.linkedin.com/jobs/view/2024-culina...   
2     https://www.linkedin.com/jobs/view/ae-sales-le...   
3     https://www.linkedin.com/jobs/view/arnp-at-fas...   
4     https://www.linkedin.com/jobs/view/assistant-m...   
...                                                 ...   
1009  https://uk.linkedin.com/jobs/view/workshop-con...   
1010  https://www.linkedin.com/jobs/view/x-ray-techn...   
1011  https://www.linkedin.com/jobs/view/xfinity-ret...   
1012  https://www.linkedin.com/jobs/view/yard-associ...   
1013  https://www.linkedin.com/jobs/view/yoga-instru...   

                last_processed_time got_summary got_ner is_being_worked  \
0      2024-01-19 14:51:42.44518+00           t       t               f   
1     2024-01-19 12:35:58.423951+00           t       t               f   
2     2024-01-19 09:45:09.215838+00           t       t               f   
3     2024-01-20 11:08:01.254837+00           t       t               f   
4     2024-01-20 03:22:05.147606+00           t       t               f   
...                             ...         ...     ...             ...   
1009  2024-01-19 09:45:09.215838+00           t       t               f   
1010  2024-01-21 04:16:54.473867+00           t       t               f   
1011  2024-01-19 23:17:04.320124+00           t       t               f   
1012  2024-01-19 09:45:09.215838+00           t       t               f   
1013  2024-01-19 09:45:09.215838+00           t       t               f   

                                              job_title  \
0     2024 - Culinary Professionals - Assistant Kitc...   
1     2024 - Culinary Professionals - Dining Hall Se...   
2                    AE - Sales Leader (Full-Time) - US   
3                                                  ARNP   
4                                     ASSISTANT MANAGER   
...                                                 ...   
1009                                Workshop Controller   
1010                                 X-Ray Technologist   
1011                    Xfinity Retail Sales Consultant   
1012                       Yard Associate (SkillBridge)   
1013                                    Yoga Instructor   

                             company  \
0                     Wolfoods, Inc.   
1                     Wolfoods, Inc.   
2     AMERICAN EAGLE OUTFITTERS INC.   
3                   Fast Pace Health   
4           d'Lite Healthy On The Go   
...                              ...   
1009          Activate Group Limited   
1010                Fast Pace Health   
1011        XFINITY Store by Comcast   
1012                ARMOR Initiative   
1013                 Active Wellness   

                                   job_location  first_seen   search_city  \
0                                Oconomowoc, WI  2024-01-14     Wisconsin   
1                                 Anchorage, AK  2024-01-15        Alaska   
2                                Providence, RI  2024-01-12     Attleboro   
3                                     Prien, LA  2024-01-14  Lake Charles   
4                                Scottsdale, AZ  2024-01-15      Chandler   
...                                         ...         ...           ...   
1009  Bury St. Edmunds, England, United Kingdom  2024-01-13       Ipswich   
1010                                  Grant, AL  2024-01-17  Guntersville   
1011                         East Brunswick, NJ  2024-01-14   Long Branch   
1012                            Kansas City, MO  2024-01-13   Leavenworth   
1013                             Plantation, FL  2024-01-13      Sarasota   

      search_country          search_position   job_level job_type  \
0      United States                Sous Chef  Mid senior   Onsite   
1      United States           Cook Fast Food  Mid senior   Onsite   
2      United States        Account Executive  Mid senior   Onsite   
3      United States       Nurse Practitio

### Sampling remaining 339 Jobs

In [14]:
#Getting Series of unique job_titles
job_titles = resp_df['job_title'].value_counts().index

In [15]:
#Gathering other job titles
title_subset = job_titles[job_titles.str.contains("Data")].tolist() + job_titles[job_titles.str.contains("Software")].tolist() + job_titles[job_titles.str.contains("Research")].tolist() + job_titles[job_titles.str.contains("Generative")].tolist() + job_titles[job_titles.str.contains("System")].tolist() + job_titles[job_titles.str.contains("UX")].tolist()

len(title_subset)

50

In [16]:
# Filter the DataFrame first
filtered_resp_df = resp_df[(resp_df['job_title'].isin(title_subset)) & (resp_df['skill_count'] >= 30) & (~resp_df['id'].isin(resp_df_max['id']))]

# Check if you have at least 339 records to sample
if len(filtered_resp_df) >= 492:
    resp_df_sample = filtered_resp_df.sample(n=492, random_state=32)
else:
    resp_df_sample = filtered_resp_df  # or handle this case as required
    print(f"Warning: Only {len(resp_df_sample)} records available; fewer than 492.")

resp_df_sample

job_link  \
34064   https://www.linkedin.com/jobs/view/software-ar...   
63847   https://www.linkedin.com/jobs/view/senior-syst...   
106562  https://www.linkedin.com/jobs/view/senior-data...   
97378   https://www.linkedin.com/jobs/view/distinguish...   
115276  https://www.linkedin.com/jobs/view/systems-adm...   
...                                                   ...   
112773  https://www.linkedin.com/jobs/view/business-sy...   
63885   https://www.linkedin.com/jobs/view/software-ar...   
52900   https://www.linkedin.com/jobs/view/principal-s...   
15817   https://uk.linkedin.com/jobs/view/lead-systems...   
57584   https://www.linkedin.com/jobs/view/senior-data...   

                  last_processed_time got_summary got_ner is_being_worked  \
34064   2024-01-19 09:45:09.215838+00           t       t               f   
63847   2024-01-19 09:45:09.215838+00           t       t               f   
106562  2024-01-20 11:15:59.804413+00           t       t               f   
97378   2024-01-19 22:25:24.801979+00           t       t               f   
115276   2024-01-21 01:06:09.90334+00           t       t               f   
...                               ...         ...     ...             ...   
112773   2024-01-20 16:33:23.92565+00           t       t               f   
63885   2024-01-19 09:45:09.215838+00           t       t               f   
52900   2024-01-19 09:45:09.215838+00           t       t               f   
15817   2024-01-19 23:23:21.954466+00           t       t               f   
57584   2024-01-19 09:45:09.215838+00           t       t               f   

                               job_title                company  \
34064                 Software Architect        Harris Computer   
63847             Senior System Engineer                 ExLabs   
106562             Senior Data Scientist  Workforce Connections   
97378   Distinguished Applied Researcher      Jobs for Humanity   
115276             Systems Administrator                 Leidos   
...                                  ...                    ...   
112773          Business Systems Analyst           SPECTRAFORCE   
63885                 Software Architect   Software People Inc.   
52900        Principal Software Engineer      Syrinx Consulting   
15817              Lead Systems Engineer              Matchtech   
57584               Senior Data Engineer             Bio-Techne   

                                job_location  first_seen    search_city  \
34064                 Florida, United States  2024-01-13  Beverly Hills   
63847                         Long Beach, CA  2024-01-13        Compton   
106562                          St Louis, MO  2024-01-14     Belleville   
97378                         Fort Worth, TX  2024-01-15     Fort Worth   
115276                  Colorado Springs, CO  2024-01-14  Cripple Creek   
...                                      ...         ...            ...   
112773                           Seattle, WA  2024-01-14         Summit   
63885                             Albany, NY  2024-01-13         Cohoes   
52900                            Bedford, MA  2024-01-13      Haverhill   
15817   Farnborough, England, United Kingdom  2024-01-14        Reading   
57584                         Centennial, CO  2024-01-13         Aurora   

        search_country                    search_position   job_level  \
34064    United States                 Director Technical  Mid senior   
63847    United States           Maintenance Data Analyst  Mid senior   
106562   United States                            Chemist  Mid senior   
97378    United States                  Director Research  Mid senior   
115276   United States                               Tier  Mid senior   
...                ...                                ...         ...   
112773   United States                    Systems Analyst  Mid senior   
63885    United States                              Guide  Mid senior   
52900    United States  Comp

In [17]:
sampled_df = pd.concat([resp_df_sample, resp_df_max], ignore_index=True)
sampled_df

job_link  \
0     https://www.linkedin.com/jobs/view/software-ar...   
1     https://www.linkedin.com/jobs/view/senior-syst...   
2     https://www.linkedin.com/jobs/view/senior-data...   
3     https://www.linkedin.com/jobs/view/distinguish...   
4     https://www.linkedin.com/jobs/view/systems-adm...   
...                                                 ...   
1495  https://uk.linkedin.com/jobs/view/workshop-con...   
1496  https://www.linkedin.com/jobs/view/x-ray-techn...   
1497  https://www.linkedin.com/jobs/view/xfinity-ret...   
1498  https://www.linkedin.com/jobs/view/yard-associ...   
1499  https://www.linkedin.com/jobs/view/yoga-instru...   

                last_processed_time got_summary got_ner is_being_worked  \
0     2024-01-19 09:45:09.215838+00           t       t               f   
1     2024-01-19 09:45:09.215838+00           t       t               f   
2     2024-01-20 11:15:59.804413+00           t       t               f   
3     2024-01-19 22:25:24.801979+00           t       t               f   
4      2024-01-21 01:06:09.90334+00           t       t               f   
...                             ...         ...     ...             ...   
1495  2024-01-19 09:45:09.215838+00           t       t               f   
1496  2024-01-21 04:16:54.473867+00           t       t               f   
1497  2024-01-19 23:17:04.320124+00           t       t               f   
1498  2024-01-19 09:45:09.215838+00           t       t               f   
1499  2024-01-19 09:45:09.215838+00           t       t               f   

                             job_title                   company  \
0                   Software Architect           Harris Computer   
1               Senior System Engineer                    ExLabs   
2                Senior Data Scientist     Workforce Connections   
3     Distinguished Applied Researcher         Jobs for Humanity   
4                Systems Administrator                    Leidos   
...                                ...                       ...   
1495               Workshop Controller    Activate Group Limited   
1496                X-Ray Technologist          Fast Pace Health   
1497   Xfinity Retail Sales Consultant  XFINITY Store by Comcast   
1498      Yard Associate (SkillBridge)          ARMOR Initiative   
1499                   Yoga Instructor           Active Wellness   

                                   job_location  first_seen    search_city  \
0                        Florida, United States  2024-01-13  Beverly Hills   
1                                Long Beach, CA  2024-01-13        Compton   
2                                  St Louis, MO  2024-01-14     Belleville   
3                                Fort Worth, TX  2024-01-15     Fort Worth   
4                          Colorado Springs, CO  2024-01-14  Cripple Creek   
...                                         ...         ...            ...   
1495  Bury St. Edmunds, England, United Kingdom  2024-01-13        Ipswich   
1496                                  Grant, AL  2024-01-17   Guntersville   
1497                         East Brunswick, NJ  2024-01-14    Long Branch   
1498                            Kansas City, MO  2024-01-13    Leavenworth   
1499                             Plantation, FL  2024-01-13       Sarasota   

      search_country           search_position   job_level job_type  \
0      United States        Director Technical  Mid senior   Onsite   
1      United States  Maintenance Data Analyst  Mid senior   Onsite   
2      United States                   Chemist  Mid senior   Onsite   
3      United States         Director Research  Mid senior   Onsite   
4      United States                      Tier  Mid senior   Onsite   
...              ...                       ...         ...      ...   
1495  United Kingdom                Controller  Mid senior   Onsite   
1496   United States   Ultrasound Technologist  Mid senior   Onsite   
1497   United States    Sales-Service Promot

In [18]:
sampled_df['id'].nunique()

1500

## Writing the sampled data

In [19]:
display(sampled_df.head(500))
display(sampled_df.head(1000).tail(500))
display(sampled_df.tail(500))

job_link  \
0    https://www.linkedin.com/jobs/view/software-ar...   
1    https://www.linkedin.com/jobs/view/senior-syst...   
2    https://www.linkedin.com/jobs/view/senior-data...   
3    https://www.linkedin.com/jobs/view/distinguish...   
4    https://www.linkedin.com/jobs/view/systems-adm...   
..                                                 ...   
495  https://www.linkedin.com/jobs/view/arnp-at-fas...   
496  https://www.linkedin.com/jobs/view/assistant-m...   
497  https://ca.linkedin.com/jobs/view/academic-adv...   
498  https://www.linkedin.com/jobs/view/account-exe...   
499  https://www.linkedin.com/jobs/view/account-exe...   

               last_processed_time got_summary got_ner is_being_worked  \
0    2024-01-19 09:45:09.215838+00           t       t               f   
1    2024-01-19 09:45:09.215838+00           t       t               f   
2    2024-01-20 11:15:59.804413+00           t       t               f   
3    2024-01-19 22:25:24.801979+00           t       t               f   
4     2024-01-21 01:06:09.90334+00           t       t               f   
..                             ...         ...     ...             ...   
495  2024-01-20 11:08:01.254837+00           t       t               f   
496  2024-01-20 03:22:05.147606+00           t       t               f   
497  2024-01-19 09:45:09.215838+00           t       t               f   
498  2024-01-19 09:45:09.215838+00           t       t               f   
499  2024-01-19 09:45:09.215838+00           t       t               f   

                                   job_title                   company  \
0                         Software Architect           Harris Computer   
1                     Senior System Engineer                    ExLabs   
2                      Senior Data Scientist     Workforce Connections   
3           Distinguished Applied Researcher         Jobs for Humanity   
4                      Systems Administrator                    Leidos   
..                                       ...                       ...   
495                                     ARNP          Fast Pace Health   
496                        ASSISTANT MANAGER  d'Lite Healthy On The Go   
497                         Academic Advisor  Mount Allison University   
498          Account Executive Digital Sales         Salem Media Group   
499  Account Executive, Industrial Equipment    Better Talent Advisors   

                         job_location  first_seen        search_city  \
0              Florida, United States  2024-01-13      Beverly Hills   
1                      Long Beach, CA  2024-01-13            Compton   
2                        St Louis, MO  2024-01-14         Belleville   
3                      Fort Worth, TX  2024-01-15         Fort Worth   
4                Colorado Springs, CO  2024-01-14      Cripple Creek   
..                                ...         ...                ...   
495                         Prien, LA  2024-01-14       Lake Charles   
496                    Scottsdale, AZ  2024-01-15           Chandler   
497  Sackville, New Brunswick, Canada  2024-01-13  Sarnia-Clearwater   
498                    Alexandria, VA  2024-01-12            Dunkirk   
499                   San Antonio, TX  2024-01-12        San Antonio   

    search_country           search_position   job_level job_type  \
0    United States        Director Technical  Mid senior   Onsite   
1    United States  Maintenance Data Analyst  Mid senior   Onsite   
2    United States                   Chemist  Mid senior   Onsite   
3    United States         Director Research  Mid senior   Onsite   
4    United States                      Tier  Mid senior   Onsite   
..             ...                       ...         ...      ...   
495  United States        Nurse Practitioner  Mid senior   Onsite   
496  United States           Complaint Clerk  Mid senior   Onsite   
497         Canada   Foreign-Student Adviser  Mid senior   Onsite   
498  United Stat

job_link  \
500  https://www.linkedin.com/jobs/view/account-exe...   
501  https://www.linkedin.com/jobs/view/account-exe...   
502  https://www.linkedin.com/jobs/view/account-sup...   
503  https://www.linkedin.com/jobs/view/accountant-...   
504  https://www.linkedin.com/jobs/view/accounting-...   
..                                                 ...   
995  https://www.linkedin.com/jobs/view/manufacturi...   
996  https://www.linkedin.com/jobs/view/marketing-c...   
997  https://uk.linkedin.com/jobs/view/marketing-ex...   
998  https://www.linkedin.com/jobs/view/marketing-s...   
999  https://www.linkedin.com/jobs/view/master-elec...   

               last_processed_time got_summary got_ner is_being_worked  \
500  2024-01-19 09:45:09.215838+00           t       t               f   
501  2024-01-19 09:45:09.215838+00           t       t               f   
502   2024-01-19 12:35:07.03423+00           t       t               f   
503  2024-01-19 09:45:09.215838+00           t       t               f   
504  2024-01-19 09:45:09.215838+00           t       t               f   
..                             ...         ...     ...             ...   
995  2024-01-21 05:25:14.377269+00           t       t               f   
996   2024-01-20 04:24:04.26509+00           t       t               f   
997  2024-01-19 09:45:09.215838+00           t       t               f   
998  2024-01-19 09:45:09.215838+00           t       t               f   
999  2024-01-19 09:45:09.215838+00           t       t               f   

                                job_title                      company  \
500  Account Executive, Industrial Supply       Better Talent Advisors   
501      Account Executive, Public Sector       Better Talent Advisors   
502                    Account Supervisor  Publicis Digital Experience   
503                         Accountant II                     Curaleaf   
504                  Accounting Assistant    Jones Grove IT Recruiting   
..                                    ...                          ...   
995              Manufacturing Supervisor          Volm Companies Inc.   
996                 Marketing Coordinator      Employee Owned Holdings   
997                   Marketing Executive             APT Travel Group   
998                  Marketing Specialist                 VP Logistics   
999                    Master Electrician                    Ascension   

                        job_location  first_seen     search_city  \
500                     Stamford, CT  2024-01-12      Bridgeport   
501                    Rochester, NY  2024-01-12      Williamson   
502                   Birmingham, MI  2024-01-16          Macomb   
503                   Scottsdale, AZ  2024-01-12        Glendale   
504                  Charlotte Metro  2024-01-12       Hollywood   
..                               ...         ...             ...   
995                       Antigo, WI  2024-01-14       Wisconsin   
996                      Houston, TX  2024-01-14         Baytown   
997  London, England, United Kingdom  2024-01-13  Greater London   
998                     Waukegan, IL  2024-01-13   North Chicago   
999                   Southfield, MI  2024-01-13        Oak Park   

     search_country       search_position   job_level job_type  \
500   United States     Account Executive  Mid senior   Onsite   
501   United States     Account Executive  Mid senior   Onsite   
502   United States    Supervisor Drawing  Mid senior   Onsite   
503   United States      Accounting Clerk  Mid senior   Onsite   
504   United States      Accounting Clerk  Mid senior   Onsite   
..              ...                   ...         ...      ...   
995   United States  Superintendent Sales  Mid senior   Onsite   
996   United States    Retirement Officer   Associate   Onsite   
997  United Kingdom       Director Social  Mid senior   Onsite   
998   United States                 Guide   Associate   Onsite   
999   United States         

job_link  \
1000  https://www.linkedin.com/jobs/view/master-sche...   
1001  https://www.linkedin.com/jobs/view/master-tech...   
1002  https://www.linkedin.com/jobs/view/material-ha...   
1003  https://www.linkedin.com/jobs/view/materials-m...   
1004  https://www.linkedin.com/jobs/view/materials-p...   
...                                                 ...   
1495  https://uk.linkedin.com/jobs/view/workshop-con...   
1496  https://www.linkedin.com/jobs/view/x-ray-techn...   
1497  https://www.linkedin.com/jobs/view/xfinity-ret...   
1498  https://www.linkedin.com/jobs/view/yard-associ...   
1499  https://www.linkedin.com/jobs/view/yoga-instru...   

                last_processed_time got_summary got_ner is_being_worked  \
1000  2024-01-21 13:08:00.689444+00           t       t               f   
1001  2024-01-19 09:45:09.215838+00           t       t               f   
1002  2024-01-21 15:59:00.747387+00           t       t               f   
1003  2024-01-21 01:30:05.765203+00           t       t               f   
1004  2024-01-19 13:06:03.847966+00           t       t               f   
...                             ...         ...     ...             ...   
1495  2024-01-19 09:45:09.215838+00           t       t               f   
1496  2024-01-21 04:16:54.473867+00           t       t               f   
1497  2024-01-19 23:17:04.320124+00           t       t               f   
1498  2024-01-19 09:45:09.215838+00           t       t               f   
1499  2024-01-19 09:45:09.215838+00           t       t               f   

                            job_title                    company  \
1000                 Master Scheduler  Baltimore Aircoil Company   
1001                Master Technician    A. O. Smith Corporation   
1002                 Material Handler   IMI Critical Engineering   
1003                Materials Manager                     Pregis   
1004                Materials Planner             Faraday Future   
...                               ...                        ...   
1495              Workshop Controller     Activate Group Limited   
1496               X-Ray Technologist           Fast Pace Health   
1497  Xfinity Retail Sales Consultant   XFINITY Store by Comcast   
1498     Yard Associate (SkillBridge)           ARMOR Initiative   
1499                  Yoga Instructor            Active Wellness   

                                   job_location  first_seen   search_city  \
1000                                 Dayton, TN  2024-01-17     Tennessee   
1001                                Lebanon, TN  2024-01-12     Watertown   
1002                 Rancho Santa Margarita, CA  2024-01-16        Corona   
1003                            Bensenville, IL  2024-01-16         Elgin   
1004                                Hanford, CA  2024-01-14         Selma   
...                                         ...         ...           ...   
1495  Bury St. Edmunds, England, United Kingdom  2024-01-13       Ipswich   
1496                                  Grant, AL  2024-01-17  Guntersville   
1497                         East Brunswick, NJ  2024-01-14   Long Branch   
1498                            Kansas City, MO  2024-01-13   Leavenworth   
1499                             Plantation, FL  2024-01-13      Sarasota   

      search_country               search_position   job_level job_type  \
1000   United States                 Traffic Agent  Mid senior   Onsite   
1001   United States                     Assembler  Mid senior   Onsite   
1002   United States              Route Supervisor   Associate   Onsite   
1003   United States  Manager Procurement Services  Mid senior   Onsite   
1004   United States                     Scheduler  Mid senior   Onsite   
...              ...                           ...         ...      ...   
1495  United Kingdom                    Controller  Mid senior   Onsite   
1496   United States       Ultrasound Technologist  Mid senior   Onsite   
1497   United States

In [20]:
# Writing to 3 different sample files
sampled_df.head(500).to_csv(f"{path}/data/sampled_linkedin_jd_skills/sampled_jd_skills_set1.csv", index=False)
sampled_df.head(1000).tail(500).to_csv(f"{path}/data/sampled_linkedin_jd_skills/sampled_jd_skills_set2.csv", index=False)
sampled_df.tail(500).to_csv(f"{path}/data/sampled_linkedin_jd_skills/sampled_jd_skills_set3.csv", index=False)